### PR TITLE
Move config to config files

### PR DESCRIPTION
**Pull request by joydeep-b**

### CODE REVIEW

File: src/navigation/navigation_main.cc [hunk 1]
```diff
@@ -62,30 +63,29 @@ using std::string;
...
-DEFINE_string(enable_topic, "autonomy_arbiter/enabled",
-    "ROS topic that indicates whether autonomy is enabled or not.");
-DEFINE_string(map,
-              "maps/Joydeepb-Home/Joydeepb-Home.navigation.txt",
-              "Name of navigation map file");
-DECLARE_string(helpon);
+const string kAmrlMapsDir = ros::package::getPath("amrl_maps");
+
+DEFINE_string(robot_config, "config/navigation.lua", "Robot config file");
+DEFINE_string(maps_dir, kAmrlMapsDir, "Robot config file");
```
Comment (kvsikand) [inline src/navigation/navigation_main.cc]: update description here

File: src/navigation/navigation_main.cc [hunk 2]
```diff
@@ -182,27 +219,32 @@ int main(int argc, char** argv) {
   // Initialize ROS.
   ros::init(argc, argv, "navigation", ros::init_options::NoSigintHandler);
   ros::NodeHandle n;
-  navigation_ = new Navigation(FLAGS_map, &n);
+  const string map_file = FLAGS_maps_dir +
+      StringPrintf("/%s/%s.navigation.txt",
+                   FLAGS_map.c_str(),
+                   FLAGS_map.c_str());
```
Comment (kvsikand) [inline src/navigation/navigation_main.cc]: abt reason this isn't using the MapNameToFile function below?

**Review by kvsikand — APPROVED** [review]
minor comments, looks good.

Eventually, is this going to replace the `navigation` folder in the `ut-jackal` repo?

File: src/navigation/navigation_main.cc [hunk 3]
```diff
@@ -62,30 +63,29 @@ using std::string;
...
-              "initialpose",
-              "Name of ROS topic for initialization");
-// Name of topic controlling whether to enable autonomous nav or not.
-DEFINE_string(enable_topic, "autonomy_arbiter/enabled",
-    "ROS topic that indicates whether autonomy is enabled or not.");
-DEFINE_string(map,
-              "maps/Joydeepb-Home/Joydeepb-Home.navigation.txt",
-              "Name of navigation map file");
-DECLARE_string(helpon);
+const string kAmrlMapsDir = ros::package::getPath("amrl_maps");
```
Comment (maxsvetlik) [inline src/navigation/navigation_main.cc:67]: An existence check could be nice here, informing the user if `amrl_maps` cannot be found.

**Review by maxsvetlik — APPROVED** [review]
I left a minor suggestion. I haven't run this to test it, but if it runs as intended it LGTM.

**Comment by kvsikand** [comment]
Opening a new one for re-review, since a lot has changed

**Pull request closed without merging**